MIDAS/src/systems.cpp

```diff
@@ -125,21 +125,27 @@ DECLARE_THREAD(buzzer, RocketSystems* arg) {
 }
 
 DECLARE_THREAD(kalman, RocketSystems* arg) {
-    yessir.initialize();
+    // Orientation initial_orientation = arg->rocket_data.orientation.getRecent();
+    // Barometer initial_barom_buf = arg->rocket_data.barometer.getRecent();
+    // LowGData initial_accelerometer = arg->rocket_data.low_g.getRecent();
+    //yessir.initialize(initial_orientation, initial_barom_buf, initial_accelerations);
+    yessir.initialize(arg);
     TickType_t last = xTaskGetTickCount();
 
     while (true) {
         // add the tick update function
-        Barometer current_barom_buf = arg->rocket_data.barometer.getRecentUnsync();
-        LowGData current_accelerometer = arg->rocket_data.low_g.getRecentUnsync();
+        Barometer current_barom_buf = arg->rocket_data.barometer.getRecent();
+        Orientation current_orientation = arg->rocket_data.orientation.getRecent();
+        LowGData current_accelerometer = arg->rocket_data.low_g.getRecent();
```

Comment:
Using the lowg isn't good here since it maxes out at 8g which is less than launch acceleration. The highg is a better choice here.

Reply:
I missed this earlier but i agree. for the future we could look at the `FSMState` and to decide which accelerometer to reaad from